src/DynamoCoreWpf/ViewModels/PackageManager/PublishPackageViewModel.cs

```diff
@@ -1067,10 +1069,33 @@ private List<PackageItemRootViewModel> BindParentToChild(Dictionary<string, Pack
             }
 
             // Only add the folder items, they contain the files
-            updatedItems = items.Values.Where(x => !x.isChild).ToList();
+            var updatedItems = GetRootItems(items);
             return updatedItems;
         }
 
+        private List<PackageItemRootViewModel> GetRootItems(Dictionary<string, PackageItemRootViewModel> items)
+        {
+            var rootItems = items.Values.Where(x => !x.isChild).ToList();
+            if (!rootItems.Any()) return rootItems;
+
+            var root = new PackageItemRootViewModel(CurrentPackageDirectory);
+            var updatedItems = new List<PackageItemRootViewModel>();
+            //check each root item and create any missing connections
+            foreach (var item in rootItems)
+            {
+                var itemDir = new DirectoryInfo(item.DirectoryName);
+                if (!itemDir.Parent.FullName.Equals(CurrentPackageDirectory))
+                {
+                    root.AddChild(item);
```

Comment:
what is the difference between these two cases of this if statement?

Reply:
btw..there are also 2 different `AddChildren` methods

Reply:
yeah, would be nice to rename is eventually as the side effect is not obvious from its name.

Reply:
Updated to `AddChildRecursively`